### ISSUE TITLE
WmsUrlBuilder does not handle multiple spaces in layer name

### ISSUE THREAD

**Issue by emxsys** [NASAWorldWind/WebWorldWind#531]
The `WmsUrlBuilder.urlForTile`  method does not replace all the spaces in layer name, only the first space.  If multiple spaces are encountered the result is an invalid URL.

https://github.com/NASAWorldWind/WebWorldWind/blob/855286bc6cc6eed4184cd6b8051f336a473d5c2a/src/util/WmsUrlBuilder.js#L158

Proposed solution: 
Use a regex expression with a `g` global flag. Example: `sb = sb.replace(/\s/g, "%20");`

**Comment by emxsys** [comment]
The issue also applies to `WcsTileUrlBuilder`.
https://github.com/NASAWorldWind/WebWorldWind/blob/855286bc6cc6eed4184cd6b8051f336a473d5c2a/src/util/WcsTileUrlBuilder.js#L117

**Comment by emxsys** [comment]
The issue applies to `LevelRowColumnUrlBuilder`
https://github.com/NASAWorldWind/WebWorldWind/blob/855286bc6cc6eed4184cd6b8051f336a473d5c2a/src/util/LevelRowColumnUrlBuilder.js#L95

**Comment by pdavidc** [comment]
Wouldn't it be better to apply a generalized URL encoder, rather than playing whack-a-mole with one problem?

**Comment by pdavidc** [comment]
Let's use the browser's builtin encodeURI function. For the moment, we'll need to apply it everywhere we perform a network request. We'll consolidate all of this in a single place with epic #154: _Network request service_.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI

**Comment by PJHogan** [comment]
Is this thing names wrong? Should URI be URL?

**Comment by pdavidc** [comment]
No, URI is the broader notion of a string indicating a network resource. URL is a type of URI.
